source/MaterialXCore/Document.h

```diff
@@ -78,13 +79,13 @@ class MX_CORE_API Document : public GraphElement
     /// Return the NodeGraph, if any, with the given name.
     NodeGraphPtr getNodeGraph(const string& name) const
     {
-        return getChildOfType<NodeGraph>(name);
+        return getChildOfTypeWithLibrary<NodeGraph>(getDataLibrary(), name);
```

Comment:
This is really starting to come together, thanks @ashwinbhat!

Looking at the latest proposal, one thought that comes to mind is that we can now merge the new functionality of `getChildOfTypeWithLibrary` directly into the existing `getChildOfType`, simply making `dataLibrary` an optional second argument rather than a mandatory first argument.

This same pattern could potentially be applied to similar helper methods in the future, where an optional `dataLibrary` argument can be passed in order to request the checking of the data library in addition to the content document.

Does that proposed simplification sound reasonable to you?

Reply:
This is what my initial intent was but I was running into some pybind issues and it was touch a lot of files. Let me try it out. I'll post an update if it works.

Reply:
With all of the latest simplifications to the proposal, I think we're now in a great place to make that change, and hopefully some combination of myself, @bernardkwok, and @ld-kerley can help to work through the Python and JS bindings!

Reply:
@jstone-lucasfilm I've simplified it. I think something like datalibrary should be a required argument but since we are trying to keep the API it should be fine. Maybe we can refine it in future.

Reply:
@ashwinbhat Here's the specific API and implementation that I was thinking we would use, and let me know if this seems like a logical approach to you:

```
    template <class T> shared_ptr<T> getChildOfType(const string& name, ConstDocumentPtr dataLibrary = nullptr) const
    {
        ElementPtr child = dataLibrary ? dataLibrary ->getChild(name) : nullptr;
        if (!child)
        {
            child = getChild(name);
        }
        return child ? child->asA<T>() : shared_ptr<T>();
    }
```

Reply:
Thanks that's what I have. I've updated the py and js bindings as well.